src/GitHub.Api/Git/Repository.cs

```diff
@@ -61,11 +140,11 @@ public void Initialize(IRepositoryManager repositoryManager)
             repositoryManager.OnRemoteBranchAdded += RepositoryManager_OnRemoteBranchAdded;
             repositoryManager.OnRemoteBranchRemoved += RepositoryManager_OnRemoteBranchRemoved;
             repositoryManager.OnGitUserLoaded += user => User = user;
-        }
 
-        public void Refresh()
-        {
-            repositoryManager?.Refresh();
+            UpdateGitStatus();
+            UpdateGitLog();
+
+            new ActionTask(CancellationToken.None, UpdateLocks) { Affinity = TaskAffinity.UI }.Start();
```

Comment:
We shouldn't be doing these three actions on Initialize. Initialize is called every time the appdomain reloads, which means every time we enter playmode we'll be doing this. We only want to do this when we first load the project in order to update information, and afterwards only when the filesystem has changes that tell us we need to do these actions.

Reply:
Where should I move it to?

Reply:
It probably needs to be something that happens when someone checks for the cache and the cache requests the data because it's either 1) too old, 2) has no data, 3) initialRun is true in ApplicationCache object. Maybe it makes sense for the cache objects to check that flag to see if they need to ask for new data when the views for ask for data? Maybe we should store a timestamp of when Unity first started up so the cache objects can check whether the cached data is older than data and request updated data when the views ask for data? Would that make sense?

Reply:
Tracking this change in https://github.com/github-for-unity/Unity/issues/409